chain-core/src/init/address.rs

```diff
@@ -26,6 +27,15 @@ use tiny_keccak::Keccak;
 
 use crate::common::{H256, HASH_SIZE_256};
 
+// CRMS: mainnet staked-state
+// CRMT: mainnet transfer
+// CRTS: testnet staked-state
+// CRTT: testnet transfer
+pub trait CroAddress<T> {
```

Comment:
Any particular reason for having this trait and not using Rust standard traits, `Display` and `FromStr`, for doing the conversion?
The advantage of implementing `Display` and `FromString` is that you get a lot of convenience functions for free. For example, you can use `.to_string()` and `.parse()` functions.

Reply:
currently, Display and FromString works for hexa address
Textual Address can vary , also can have multiple forms like container.

Reply:
I think these standard traits would be possible, i.e. change their current `impl` in corresponding addresses.
The only drawback would be that it'd require conditional compilation (as they don't take extra arguments) -- https://github.com/crypto-com/chain/pull/187#discussion_r301376681

This requirement would then propagate downstream -- e.g. it wouldn't be possible / easy to manage testnet and mainnet wallets inside client-* at the same time

Reply:
how about standard traits for CroAddress?

Reply:
One more advantage of using standard traits is that `client-cli` and `client-rpc` already uses them to display data. If we use a custom trait, we'll have to search for every usage of `.to_string()` and `.parse()` for any address type and change it to use `CroAddress`.

Reply:
One thing to watch out there though is that the hex-encoded variant is still needed for RedeemAddress in InitConfig

Reply:
> One thing to watch out there though is that the hex-encoded variant is still needed for RedeemAddress in InitConfig

Why can't we change that to the new format?

Reply:
We can -- but InitConfig is, from a large part, a textual dump of Ethereum ERC20 state and all the Eth account-related information published in official materials.

Reply:
hence the hex encoding would preferable stay in InitConfig, as it'll be easier for verification

Reply:
OK,
let's use standard trait.
I'll work on it.